### PR TITLE
Add factory methods for TransformerSuppliers

### PR DESCRIPTION
**Pull request by philipp94831**

### CODE REVIEW

File: src/test/java/com/bakdata/kafka/ErrorCapturingFlatValueTransformerTopologyTest.java [hunk 1]
```diff
@@ -58,7 +59,7 @@
     protected void buildTopology(final StreamsBuilder builder) {
         final KStream<Integer, String> input = builder.stream(INPUT_TOPIC, Consumed.with(null, STRING_SERDE));
         final KStream<Integer, ProcessedValue<String, Long>> mapped =
-                input.flatTransformValues(() -> ErrorCapturingFlatValueTransformer.captureErrors(this.mapper));
+                input.flatTransformValues(ErrorCapturingFlatValueTransformer.captureErrors(() -> this.mapper));
```
Comment (torbsto) [inline src/test/java/com/bakdata/kafka/ErrorCapturingFlatValueTransformerTopologyTest.java:62]: The removed line still works, right? Why remove its test case?

File: src/main/java/com/bakdata/kafka/ErrorCapturingFlatValueTransformer.java [hunk 2]
```diff
@@ -89,6 +91,55 @@
...
+     *
+     * @param supplier {@code ValueTransformerSupplier} whose exceptions should be captured
+     * @param <V> type of input values
+     * @param <VR> type of output values
+     * @return {@code ValueTransformerSupplier}
+     * @see #captureErrors(ValueTransformerSupplier, Predicate)
+     * @see ErrorUtil#isRecoverable(Exception)
+     */
+    public static <V, VR> ValueTransformerSupplier<V, Iterable<ProcessedValue<V, VR>>> captureErrors(
+            final ValueTransformerSupplier<? super V, ? extends Iterable<VR>> supplier) {
```
Comment (torbsto) [inline src/main/java/com/bakdata/kafka/ErrorCapturingFlatValueTransformer.java:106]: `@NonNull` missing. Also applies to all other `captureErrors` methods.

Comment (philipp94831) [reply src/main/java/com/bakdata/kafka/ErrorCapturingFlatValueTransformer.java]: Done

File: src/main/java/com/bakdata/kafka/ErrorCapturingFlatValueTransformer.java [hunk 3]
```diff
@@ -89,6 +91,55 @@
...
+     *
+     * @param supplier {@code ValueTransformerSupplier} whose exceptions should be captured
+     * @param errorFilter expression that filters errors which should be thrown and not captured
+     * @param <V> type of input values
+     * @param <VR> type of output values
+     * @return {@code ValueTransformerSupplier}
+     */
+    public static <V, VR> ValueTransformerSupplier<V, Iterable<ProcessedValue<V, VR>>> captureErrors(
+            final @NonNull ValueTransformerSupplier<? super V, ? extends Iterable<VR>> supplier,
+            final Predicate<Exception> errorFilter) {
```
[the remark this answers is not in the document]
Comment (philipp94831) [reply src/main/java/com/bakdata/kafka/ErrorCapturingFlatValueTransformer.java]: Done

**Review by torbsto — APPROVED** [review]

**Pull request merged**